File: Java/Hasta Entraga 3/irrgarten/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alejandro
 */
public class Shield {
    private float protection;
    private int uses;
    
    // Constructor por defecto
    public Shield(float protection,int uses){
        this.protection = protection;
        this.uses = uses;
    }
    
    // Método que devuelve la defensa del arma y reduce 1 a sus usos
    // En caso de que los usos sean 0 devuelve cero.
    public float protec(){
        float resultado = 0.0f;
        
        if(uses > 0){
            uses--;
            resultado = protection;
        }
        
        return resultado;
    }
    
    // Método que devulve un string coné los usos y defensa del arma
    @Override
    public String toString(){
        return "W[" + protection + ", " + uses + "]";
    }
    
    /*
      Este método produce el resultado delegando en el método discardElement
      del dado pasando al mismo la cantidad de usos restantes como parámetr
    */
    public boolean discard(){
        return Dice.discardElement(uses);
    }
}

File: Ruby/Hasta Entrega 3/Irrgarten/Irrgarten/shield.rb
require_relative "dice"

module Irrgarten
  class Shield
    def initialize(protection, uses)
      @protection = protection
      @uses = uses
    end

    def protec
      if @uses > 0
        @uses -= 1
        @protection
      else
        0.0
      end
    end

    def discard
      Dice.discard_element(@uses)
    end

    def to_s
      "S[#{@protection}, #{@uses}]"
    end
  end
end
